anima/env/mayaEnv/__init__.py

```diff
@@ -439,19 +450,47 @@ def save_as(self, version, run_pre_publishers=True):
 
         # run post publishers here
         if version.is_published:
+            from anima.ui.lib import QtCore, QtWidgets, QtGui
+
             # before doing anything run all publishers
             type_name = ''
             if version.task.type:
                 type_name = version.task.type.name
 
+            # show splash screen during post publish progress and lock maya
```

Comment:
I'm not keen about this part. If you think that it is really needed, you should show a proper stylized PySide2 dialog instead of an image.

Reply:
I remembered why I used an image instead of a dialog here. It was because users always tend to close the dialog without reading what it says. One might think that a frameless dialog without a close button will do the trick but the dialog still can be closed from windows taskbar. I even caught someone trying to close it from task manager :).

However using an image might not be the best solution here as you mentioned so how about adding this function below in anima.ui.utils. Once the dialog is initialized it can be showed or closed whenever needed with default show(), close() functions... in this case within publish_checker.py...

def initialize_post_publish_dialog():
    """
    A frameless, staysOnTop dialog to be initialized during post publish process
    for locking its application and actually show user that post publishes are in progress.
    """
    try:
        _fromUtf8 = QtCore.QString.fromUtf8
    except AttributeError:
        _fromUtf8 = lambda s: s

    dialog = QtWidgets.QDialog()
    dialog.vertical_layout = QtWidgets.QVBoxLayout(dialog)
    dialog.label = QtWidgets.QLabel(dialog.vertical_layout.widget())
    dialog.label.setText('POST PUBLISH IN PROGRESS')
    dialog.label.setStyleSheet(_fromUtf8("color: rgb(20, 255, 20);\n""font: 16pt;"))
    dialog.vertical_layout.addWidget(dialog.label)
    dialog.label1 = QtWidgets.QLabel(dialog.vertical_layout.widget())
    dialog.label1.setText('PLEASE WAIT...')
    dialog.label1.setAlignment(QtCore.Qt.AlignCenter)
    dialog.vertical_layout.addWidget(dialog.label1)
    dialog.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint)
    dialog.setWindowFlags(QtCore.Qt.FramelessWindowHint)
    dialog.setWindowModality(QtCore.Qt.ApplicationModal)

    return dialog


what do you say?